src/confcom/azext_confcom/template_util.py

```diff
@@ -1494,3 +1494,185 @@ def pick(obj, *keys):
+        "fragments": [],  # empty by default in your example
+        "containers": []
+    }
+
+    old_containers = old_data.get("containers", [])
+
+    for old_container in old_containers:
+        # Build the 'environmentVariables' section in the new format
+        new_envs = []
+        for env_var in old_container.get("environmentVariables") or []:
+            # Decide if we need 'regex' or not, based on 'strategy' or your custom logic
+            # Here we'll assume "strategy"=="re2" means 'regex' = True
+            # If strategy is missing or 'string', omit 'regex' or set it to False
+            env_entry = {
+                "name": env_var.get("name"),
+                "value": env_var.get("value", "")
+            }
+            strategy = env_var.get("strategy")
+            if strategy == "re2":
+                env_entry["regex"] = True
+
+            new_envs.append(env_entry)
+
+        # Build the 'execProcesses' from the old 'command'
+        exec_processes = []
+        old_command_list = old_container.get("execProcesses", [])
+        if old_command_list:
+            exec_processes.append({"command": old_command_list})
+
+        command = old_container.get("command")
+
+        # Liveness probe => exec process
+        liveness_probe = old_container.get("livenessProbe", {})
+        liveness_exec = liveness_probe.get("exec", {})
+        liveness_command = liveness_exec.get("command", [])
+        if liveness_command:
+            exec_processes.append({
+                "command": liveness_command
+            })
+
+        # Readiness probe => exec process
+        readiness_probe = old_container.get("readinessProbe", {})
+        readiness_exec = readiness_probe.get("exec", {})
+        readiness_command = readiness_exec.get("command", [])
+        if readiness_command:
+            exec_processes.append({
+                "command": readiness_command
+            })
+
+
+        # Build the 'volumeMounts' section
+        volume_mounts = []
+        for mount in old_container.get("mounts") or []:
+            # For 'name', we can take the mountType or generate something else:
+            # e.g. if mountType is "azureFile", name "azurefile"
+            mount_name = mount.get("mountType", "defaultName").lower()
+            volume_mount = {
+                "name": mount_name,
+                "mountPath": mount.get("mountPath"),
+                "mountType": mount.get("mountType"),
+                "readOnly": mount.get("readonly", True),
+            }
+            volume_mounts.append(volume_mount)
+
+        # Create the container's "properties" object
+        container_properties = {
+            "image": old_container.get("containerImage"),
+            "execProcesses": exec_processes,
+            "volumeMounts": volume_mounts,
+            "environmentVariables": new_envs,
+            "command": command,
+        }
+
+        if old_container.get("securityContext") is not None:
```

Comment:
Are these values ('securityContext' etc) not available in the config file? ie. config.ACI_FIELD_SECURITY_CONTEXT?

Reply:
They are. Initially we split out some of the fields that are in `securityContext` into their own fields because we weren't sure what ACI's format would be. In the new format, `securityContext` takes its format from the ACI docs. I'm not sure if that answers the question